hazelcast-client/src/test/java/com/hazelcast/client/ClientRegressionWithRealNetworkTest.java

```diff
@@ -91,4 +94,22 @@ public void run() {
 
         assertOpenEventually(clientLatch);
     }
+
+    @Test
+    public void testConnectWithDNSHostnames() throws InterruptedException {
+        Config config = new Config();
+        config.getNetworkConfig().setPublicAddress("localhost");
+        HazelcastInstance hazelcastInstance = Hazelcast.newHazelcastInstance(config);
+
+        ClientConfig clientConfig = new ClientConfig();
+        clientConfig.getNetworkConfig().addAddress("localhost");
+        ClientStateListener clientStateListener = new ClientStateListener(clientConfig);
+        HazelcastClient.newHazelcastClient(clientConfig);
+
+        Hazelcast.newHazelcastInstance(config);
+
+        hazelcastInstance.shutdown();
+
+        assertTrue(clientStateListener.awaitConnected(5, TimeUnit.SECONDS));
```

Comment:
I think this line should be before `hazelcastInstance.shutdown`. Client could be decided to shutdown before assert. In this case test will fail. 
Actually given that we have proper cleanups, we don't need `shutdown` here at all.

Reply:
The test forces the client to disconnect from the owner member, that is why I need that instance shutdown. However, considering your comment, I did increase the connection attempt limit to inifinite:
clientConfig.getNetworkConfig().addAddress("localhost").setConnectionAttemptLimit(Integer.MAX_VALUE);